chia/server/class_for_type.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+
+from typing import cast
+
+from chia.farmer.farmer_api import FarmerAPI
+from chia.full_node.full_node_api import FullNodeAPI
+from chia.harvester.harvester_api import HarvesterAPI
+from chia.introducer.introducer_api import IntroducerAPI
+from chia.server.api_protocol import ApiProtocol
+from chia.server.outbound_message import NodeType
+from chia.timelord.timelord_api import TimelordAPI
+from chia.wallet.wallet_node_api import WalletNodeAPI
+
+
+def register_all(d: dict[NodeType, ApiProtocol]) -> None:
+    for k, v in [
+        (NodeType.FULL_NODE, FullNodeAPI),
+        (NodeType.WALLET, WalletNodeAPI),
+        (NodeType.INTRODUCER, IntroducerAPI),
+        (NodeType.TIMELORD, TimelordAPI),
+        (NodeType.FARMER, FarmerAPI),
+        (NodeType.HARVESTER, HarvesterAPI),
+    ]:
+        d[k] = cast(ApiProtocol, v)
+
+
+GLOBAL_REGISTRY: dict[NodeType, ApiProtocol] = {}
+
+register_all(GLOBAL_REGISTRY)
+
+class_for_type: dict[NodeType, ApiProtocol] = GLOBAL_REGISTRY
```

Comment:
Is there a reason that led to this indirection? in my local experiment I have this passed as a constant dict (similar to this `GLOBAL_REGISTRY` after `register_all`) and that works too.

Reply:
I think I followed enough history here to see how it got here, but it does seem like it could be simplified another step.

Is this the concrete suggestion here?

```suggestion
class_for_type: dict[NodeType, ApiProtocol] = {
    NodeType.FULL_NODE: FullNodeAPI,
    NodeType.WALLET: WalletNodeAPI,
    NodeType.INTRODUCER: IntroducerAPI,
    NodeType.TIMELORD: TimelordAPI,
    NodeType.FARMER: FarmerAPI,
    NodeType.HARVESTER: HarvesterAPI,
}
```

On the flip side, if we kept the registry/registration concept we could eliminate all the imports from here to the APIs and instead have the APIs import this file and register themselves.  `@register_api(NodeType.FULL_NODE)` as a decorator for `class FullNodeAPI:` for example.